main.go

```diff
@@ -79,6 +80,7 @@ func (s *SuperAgent) ClearSuperAgent() {
 	s.TargetType = "json"
 	s.Cookies = make([]*http.Cookie, 0)
 	s.Errors = nil
+	s.BasicAuth = nil
```

Comment:
does this fix the issue with reuse you mentioned? You mentioned something about the `New()` function, but I don't see how there would be an issue of reuse there